Include/cpython/pystats.h

```diff
@@ -100,10 +96,34 @@ typedef struct _gc_stats {
     uint64_t objects_collected;
 } GCStats;
 
+typedef struct _uop_stats {
+    uint64_t execution_count;
+} UOpStats;
+
+#define _Py_UOP_HIST_SIZE 5
```

Comment:
We'll probably want more buckets than this (since execution lengths for closed loops will get pretty big), but it's probably fine for now. We can just bump it whenever.

Reply:
> Can they be longer? (It's quite possible I don't understand the code).

Not statically, but dynamically they can be quite long (imagine executing a trace for a closed loop hundreds of times).

So we should probably modify the code that adds to the histogram to be `Py_MAX(_Py_UOP_HIST_SIZE - 1, _Py_bit_length((length - 1) >> 2))]`, so that anything overflowing the histogram ends up in the largest bucket instead of writing past the end of the array.

Reply:
Yeah, that's a good idea to add the range-checking.

Also, this made me realize I was counting the wrong thing -- I was recording the "instruction pointer at exit" to be the execution length.  But (after discussing in person) it seems what we want is the number of uops executed per execution of the trace.  I've updated this to report that instead.